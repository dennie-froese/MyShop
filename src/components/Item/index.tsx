import React, {useState} from 'react';
import {View, StyleSheet, Text, Image, TouchableOpacity} from 'react-native';
import {useIncDec} from 'hooks/useTotalContext';

export type Product = {
  id: number;
  colour: string;
  name: string;
  price: number;
  img: string;
};
interface Props {
  product: Product;
  remove: (id: number) => void;
}

export default function Item({product, remove}: Props) {
  const [itemCount, setItemCount] = useState(0);
  const {inc, dec} = useIncDec(product.price);
  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image style={styles.image} source={{uri: product.img}} />
      </View>
      <View style={styles.containerRight}>
        <View>
          <Text>{product.name}</Text>
        </View>
        <View style={styles.details}>
          <View style={styles.priceContainter}>
            <Text>Â£{product.price}</Text>
          </View>
          <View style={styles.containerRightRight}>
            <View style={styles.containerQuantity}>
              <TouchableOpacity
                onPress={() => {
                  inc(), setItemCount((prev) => prev + 1);
                }}
                style={styles.increment}>
                <Text style={styles.text}>+</Text>
              </TouchableOpacity>
              <Text style={styles.text}>{itemCount}</Text>
              <TouchableOpacity
                onPress={() => {
                  dec(), setItemCount((prev) => prev - 1);
                }}
                style={styles.increment}>
                <Text style={styles.text}>-</Text>
              </TouchableOpacity>
            </View>
            <TouchableOpacity onPress={() => remove(product.id)}>
              <Text style={styles.text}>Remove</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    padding: 30,
  },
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
  image: {
    width: 65,
    height: 65,
  },
  containerRight: {
    flex: 1,
  },
  priceContainter: {
    flex: 1,
    justifyContent: 'center',
    paddingLeft: 20,
  },
  containerRightRight: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  details: {
    flex: 1,
    flexDirection: 'row',
  },
  containerQuantity: {
    flexDirection: 'row',
    alignContent: 'space-between',
  },
  text: {
    textAlign: 'center',
  },
  increment: {
    width: 30,
  },
});
