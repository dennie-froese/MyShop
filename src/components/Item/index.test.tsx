import TestRenderer from 'react-test-renderer';
import React from 'react';
import Item from '.';

const product = {
  id: 10,
  colour: 'blue',
  name: 'Top',
  price: 7,
  img: 'string',
};

describe('Item', () => {
  it('renders', () => {
    expect(
      TestRenderer.create(
        <Item product={product} increment={jest.fn()} remove={jest.fn()} />,
      ).toJSON(),
    ).toMatchSnapshot();
  });
});
