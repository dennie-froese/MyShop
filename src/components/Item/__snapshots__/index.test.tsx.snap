// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "padding": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "string",
        }
      }
      style={
        Object {
          "height": 65,
          "width": 65,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text>
        Top
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingLeft": 20,
          }
        }
      >
        <Text>
          7
           Â£
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "space-between",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "width": 30,
              }
            }
          >
            <Text
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              +
            </Text>
          </View>
          <Text
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            0
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "width": 30,
              }
            }
          >
            <Text
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              -
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Remove
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
